### PR TITLE
Render tag attributes depending on runtime variable value

### LINKED ISSUE #34: No way to conditionally add attribute?
I want to either set the "checked" attribute of a checkbox, or to have no such attribute at all. Because if the attribute is present but the value is e.g. an empty string, the box is still considered to be checked.

Not sure how to do this with slim_fast. Tried things like:

```
# Compilation error
input type="checkbox" name="meta" value="true" checked=(if @meta, do: "true")

# Not rendered correctly at all
input*[ type: "checkbox", name: "Meta", value: "true", checked: @meta ]
```

Ended up working around it by rendering the entire element in two different logic branches:

``` slim
      = if @meta do
        input type="checkbox" name="meta" value="true" checked="checked"

      = unless @meta do
        input type="checkbox" name="meta" value="true"
```

### PR DESCRIPTION
**Pull request by Rakoth**
Resolve #34
@doomspork  I am thinking of some thing like this to resolve issue with attribute value. I is not possible to evaluate all values at compile time, because we can call template with different variables in runtime.

### CODE REVIEW

**Comment by Rakoth** [comment]
We can use `(fun -> ... end).()` wrapping to not create any variables in template context

File: lib/slim_fast/compiler.ex [hunk 1]
```diff
@@ -12,10 +12,10 @@ defmodule SlimFast.Compiler do
   defp render_attribute(name, {:eex, opts}) do
     value = opts[:content]
     case value do
-      "true"  -> name
+      "true" -> " #{to_string(name)}"
       "false" -> ""
-      "nil"   -> ""
-      _       -> ~s(#{to_string(name)}="<%=#{value}%>")
+      "nil" -> ""
+      _ ->  ~s[<%= __k = "#{to_string(name)}"; __v = #{value}; if __v == true, do: " " <> __k, else: (if __v, do: ~s( \#{__k}="\#{__v}")) %>]
```
Comment (doomspork) [inline lib/slim_fast/compiler.ex:18]: This is really quite clever @Rakoth :+1:

**Comment by doomspork** [comment]
@Rakoth this looks good to me, thanks for knocking this one out.  I'm good to merge this if you are.

**Comment by Rakoth** [comment]
@doomspork go ahead :ship:

**Pull request merged**